packages/sanity/src/router/RouterProvider.tsx

```diff
@@ -119,22 +121,24 @@ export function RouterProvider(props: RouterProviderProps): React.JSX.Element {
     [routerProp, state],
   )
 
-  const navigate = useCallback(
-    (nextState: Record<string, unknown> | null, options: NavigateOptions = {}) => {
-      const currentParams = Array.isArray(state._searchParams) ? state._searchParams : []
+  const navigate: RouterContextValue['navigate'] = useCallback(
+    (nextStateOrOptions: NextStateOrOptions, maybeOptions?: NavigateOptions) => {
+      // Determine options and state based on input pattern
+      const isOptionsOnlyPattern = isNavigateOptions(nextStateOrOptions) && !maybeOptions
```

Comment:
same as above re checking arity – not a blocker for merge since it doesn't affect the public api, so can improve this later on